net/src/parse.rs

```diff
@@ -0,0 +1,167 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Open Network Fabric Authors
+
+//! Packet parsing traits
+#![allow(missing_docs)] // temporary allowance (block merge)
+
+use std::num::NonZero;
+
+pub trait Parse: Sized {
+    type Error: core::error::Error;
+    /// Parse from a buffer.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error in the event that parsing fails.
+    fn parse(buf: &[u8]) -> Result<(Self, NonZero<usize>), ParseError<Self::Error>>;
+}
+
+pub trait DeParse {
+    type Error;
+
+    fn size(&self) -> NonZero<usize>;
+    /// Write a data structure (e.g., a packet header) to a buffer.
+    ///
+    /// Returns the number of bytes written in the event of success.
+    ///
+    /// # Errors
+    ///
+    /// Will return an error if there is not enough space in the buffer
+    /// or if serialization fails from some other (implementation-dependent) reason.
+    fn write(&self, buf: &mut [u8]) -> Result<NonZero<usize>, DeParseError<Self::Error>>;
```

Comment:
rename this to deparse to avoid trait function conflicts, just like we do in Parse.  serde uses serialize, so we don't want that, and tons of people have read.

Reply:
Yup.  Will rafactor